src/components/Microsoft.DotNet.UpgradeAssistant.Analysis/AnalyzePackageStatus.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.UpgradeAssistant.Dependencies;
+using Microsoft.DotNet.UpgradeAssistant.Steps.Packages;
+using Microsoft.DotNet.UpgradeAssistant.Steps.Solution;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Analysis
+{
+    public class AnalyzePackageStatus : IAnalyzeResultProvider
+    {
+        private readonly IDependencyAnalyzerRunner _packageAnalyzer;
+        private readonly IEntrypointResolver _entrypointResolver;
+        private readonly UpgradeOptions _options;
+
+        private IDependencyAnalysisState? _analysisState;
+
+        protected ILogger Logger { get; }
+
+        public AnalyzePackageStatus(UpgradeOptions options,
+            IDependencyAnalyzerRunner packageAnalyzer,
+            ILogger<AnalyzePackageStatus> logger,
+            IEntrypointResolver entrypointResolver)
+        {
+            Logger = logger;
+            _options = options;
+            _packageAnalyzer = packageAnalyzer ?? throw new ArgumentNullException(nameof(packageAnalyzer));
+            _entrypointResolver = entrypointResolver ?? throw new ArgumentNullException(nameof(entrypointResolver));
+            _analysisState = null;
+        }
+
+        public async Task AnalyzeAsync(AnalyzeContext analysis, CancellationToken token)
+        {
+            if (analysis is null)
+            {
+                throw new ArgumentNullException(nameof(analysis));
+            }
+
+            var context = analysis.UpgradeContext;
+            var projects = context.Projects.ToList();
+
+            if (!context.InputIsSolution)
+            {
+                var selectedProject = projects.First(i => i.FileInfo.Name.Equals(Path.GetFileName(context.InputPath), StringComparison.OrdinalIgnoreCase));
+                context.EntryPoints = new[] { selectedProject };
+            }
+            else
+            {
+                context.EntryPoints = _entrypointResolver.GetEntrypoints(context.Projects, _options.EntryPoint.Any() ? _options.EntryPoint : new[] { "*" });
+            }
+
+            foreach (var project in context.EntryPoints)
```

Comment:
if this is part of an iterative approach this makes sense for now but would prefer to see that called out to the user

Reply:
We should have the `IUpgradeContext.Projects` return all the projects that are part of the chain from the entrypoint (in the specified order as well). Then we could rely on that throughout the codebase for the recommended order and the projects we actually care about.

Reply:
Changed to analyze all projects instead of just the entrypoint projects. Created https://github.com/dotnet/upgrade-assistant/issues/556 to track the change to IUpgradeContext.Projects.